src/replacement/extensions.rs

```diff
@@ -568,6 +569,85 @@ pub trait FromPathExt {
         Self: Sized;
 }
 
+pub trait FromFile {
+    fn from_file(file: &File) -> Self;
+}
+
+pub trait FromFolder {
+    fn from_folder(folder: &Folder) -> Self;
+}
```

Comment:
I find the name of these traits super misleading. Could you perhaps use a name that doesn't imply we're literally manipulating a File instance? FilePath would've worked just fine and not conflict with std::fs::File.

Reply:
I think that the names `File` and `Folder` are fine, I will gladly swap them out for `search::File` and `search::Folder` here, but `File` and `Folder` are fine imo for what they represent. If you glob import on `arc_config::search::*` then that's on you? Names conflict all the time I don't see a reason to further qualify these names, other than using `search::File` and `search::Folder`.

IF you are just talking about the trait names then I'll change them to `FromSearchFile` and `FromSearchFolder`, or maybe I'll just implement `From<&search::File>` instead of a custom trait for no reason.